Rerun analyses

komalsrathi opened the pull request:
This PR reruns all analyses with v1 release. 

Fixes https://github.com/d3b-center/hope-cohort-analysis/issues/77, https://github.com/d3b-center/hope-cohort-analysis/issues/52

jharenza commented:
hi @komalsrathi can you merge main/master into this? I think you are missing files in your survival module for eg

komalsrathi commented:
I did that but might have done something weird if you are not seeing those files. I don't know if I deleted them by mistake.

```
(base) ➜  d3b-hope-cohort-analysis git:(rerun-analyses) git checkout master
Switched to branch 'master'

(base) ➜  d3b-hope-cohort-analysis git:(master) ✗ git pull origin master
From github.com:d3b-center/d3b-hope-cohort-analysis
 * branch            master     -> FETCH_HEAD
Already up to date.

(base) ➜  d3b-hope-cohort-analysis git:(master) ✗ git checkout rerun-analyses
Switched to branch 'rerun-analyses'

(base) ➜  d3b-hope-cohort-analysis git:(rerun-analyses) git merge origin/master
Already up to date.
```

komalsrathi commented:
I'll check what is missing and add those files back from master.

komalsrathi commented:
Could you tell me which files are missing?

branch `master` has 4 directories and 43 files:

```
(base) ➜  d3b-hope-cohort-analysis git:(master) ✗ tree analyses/survival-analysis
analyses/survival-analysis
├── 01-survival-analysis.R
├── 02-summary-statics-TP53.Rmd
├── 02-summary-statics-TP53.html
├── 03-survival-tp53.Rmd
├── 03-survival-tp53.html
├── README.md
├── input
│   ├── GBM_survial_tp53.tsv
│   └── HGG_survial_tp53.tsv
├── plots
│   ├── HR_DMG.pdf
│   ├── HR_DMG_tp53_interaction.pdf
│   ├── HR_HGG_WT.pdf
│   ├── HR_broad_mol_subtype.pdf
│   ├── HR_mol_subtype_without_tp53.pdf
│   ├── HR_molecular_subtype.pdf
│   ├── HR_race.pdf
│   ├── KM_binned_age.pdf
│   ├── KM_binned_tp53.pdf
│   ├── KM_broad_mol.pdf
│   ├── KM_mol_subtype_without_tp53.pdf
│   ├── KM_race.pdf
│   ├── KM_reported_gender.pdf
│   ├── boxplot_tp53_score_binned_age_2gp_aya.pdf
│   ├── boxplot_tp53_score_binned_age_3gp_aya.pdf
│   ├── boxplot_tp53_score_gender.pdf
│   ├── boxplot_tp53_score_race.pdf
│   ├── boxplot_tp53_score_status.pdf
│   └── boxplot_tp53_score_subtype.pdf
├── results
│   ├── DMG_score+reported_gender+molecular_subtype*tp53+binned_age.RDS
│   ├── DMG_score+reported_gender+molecular_subtype+binned_age.RDS
│   ├── HGG_H3WT_score+reported_gender+molecular_subtype+binned_age.RDS
│   ├── alt_status_vs_survival_multivariate.pdf
│   ├── alt_vs_survival_km.pdf
│   ├── msi_paired_vs_survival_multivariate.pdf
│   ├── msi_status_vs_survival_km.pdf
│   ├── msi_status_vs_survival_multivariate.pdf
│   ├── t_n_telomere_content_vs_survival_multivariate.pdf
│   ├── tp53_score+reported_gender+broad_mol+binned_age.RDS
│   ├── tp53_score+reported_gender+mol_subtype+binned_age.RDS
│   ├── tp53_score+reported_gender+mol_subtype_without_tp53+binned_age.RDS
│   └── tp53_score+reported_gender+race+binned_age.RDS
└── utils
    ├── plotForest.R
    ├── survival_models.R
    └── theme_for_plot.R

4 directories, 43 files

```

branch `rerun-analyses` has 4 directories and 43 files:

```
(base) ➜  d3b-hope-cohort-analysis git:(rerun-analyses) tree analyses/survival-analysis
analyses/survival-analysis
├── 01-survival-analysis.R
├── 02-summary-statics-TP53.Rmd
├── 02-summary-statics-TP53.html
├── 03-survival-tp53.Rmd
├── 03-survival-tp53.html
├── README.md
├── input
│   ├── GBM_survial_tp53.tsv
│   └── HGG_survial_tp53.tsv
├── plots
│   ├── HR_DMG.pdf
│   ├── HR_DMG_tp53_interaction.pdf
│   ├── HR_HGG_WT.pdf
│   ├── HR_broad_mol_subtype.pdf
│   ├── HR_mol_subtype_without_tp53.pdf
│   ├── HR_molecular_subtype.pdf
│   ├── HR_race.pdf
│   ├── KM_binned_age.pdf
│   ├── KM_binned_tp53.pdf
│   ├── KM_broad_mol.pdf
│   ├── KM_mol_subtype_without_tp53.pdf
│   ├── KM_race.pdf
│   ├── KM_reported_gender.pdf
│   ├── boxplot_tp53_score_binned_age_2gp_aya.pdf
│   ├── boxplot_tp53_score_binned_age_3gp_aya.pdf
│   ├── boxplot_tp53_score_gender.pdf
│   ├── boxplot_tp53_score_race.pdf
│   ├── boxplot_tp53_score_status.pdf
│   └── boxplot_tp53_score_subtype.pdf
├── results
│   ├── DMG_score+reported_gender+molecular_subtype*tp53+binned_age.RDS
│   ├── DMG_score+reported_gender+molecular_subtype+binned_age.RDS
│   ├── HGG_H3WT_score+reported_gender+molecular_subtype+binned_age.RDS
│   ├── alt_status_vs_survival_multivariate.pdf
│   ├── alt_vs_survival_km.pdf
│   ├── msi_paired_vs_survival_multivariate.pdf
│   ├── msi_status_vs_survival_km.pdf
│   ├── msi_status_vs_survival_multivariate.pdf
│   ├── t_n_telomere_content_vs_survival_multivariate.pdf
│   ├── tp53_score+reported_gender+broad_mol+binned_age.RDS
│   ├── tp53_score+reported_gender+mol_subtype+binned_age.RDS
│   ├── tp53_score+reported_gender+mol_subtype_without_tp53+binned_age.RDS
│   └── tp53_score+reported_gender+race+binned_age.RDS
└── utils
    ├── plotForest.R
    ├── survival_models.R
    └── theme_for_plot.R

4 directories, 43 files
```

jharenza commented:
I see them now, I am not sure what happened- looks like they were updated around 4 pm!

zzgeng reviewed (commented):
Code works fine! I leave a few comments. Besides the comments in scripts, do we also need to update the bash script `[run_full_analysis.sh](https://github.com/d3b-center/hope-cohort-analysis/blob/master/run_full_analysis.sh)`?

komalsrathi commented on analyses/alt-analysis/01-correlation_alt_vs_vars.R:
```diff
@@ -1,121 +1,227 @@
 # output directory
 root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
-input_dir <- file.path(root_dir, "analyses", "master-annotation", "results")
+data_dir <- file.path(root_dir, "data", "v1")
```
Sure, let me remove the version number from the scripts.

komalsrathi commented on analyses/README.md:
```diff
@@ -0,0 +1,13 @@
+7. `survival-analysis`: Survival analysis with `ALT status` and `Molecular subtypes`.
+8. `oncoplots`: Oncoplots and cascade style plots.
+9. `proteomic-data`:
+10. `tp53_nf1_score`:
```
Okay got it, will do.

komalsrathi commented:
> Code works fine! I leave a few comments. Besides the comments in scripts, do we also need to update the bash script `[run_full_analysis.sh](https://github.com/d3b-center/hope-cohort-analysis/blob/master/run_full_analysis.sh)`?

I removed this script as it isn't needed anymore.

jharenza commented on analyses/oncoplots/02-prepare_files_oncogrid.R:
```diff
@@ -103,8 +103,9 @@ annot_info <- hist_df %>%
-  dplyr::select(Sample, Sequencing_Experiment, Diagnosis, Diagnosis_Type, Tumor_Location, Sex, Age, TMB) %>%
+                "Age" = "HARMONY_age_class_derived",
+                "Molecular_Subtype" = "molecular_subtype") %>%
+  dplyr::select(Sample, Sequencing_Experiment, Diagnosis, Molecular_Subtype, Diagnosis_Type, Tumor_Location, Sex, Age, TMB) %>%
```
hi @komalsrathi line 104, can you make that `CNS_region` instead?  Also trying to get v2 out today so we will update instead of `molecular_subtype` using `cancer_group_short`... will let you know when that's done.

jharenza replied:
cc @mkoptyra

komalsrathi replied:
Just saw this - somehow missed it last week. 

1. So for the oncoplots, we don't need `molecular_subtype` in the annotation, but instead we want `cancer_group_short`? I also have `Diagnosis` (which is a combination of `HOPE_diagnosis` + `HARMONY_WHO.Grade`) in the annotation, should I keep it or remove it? 

2. Also, should `Tumor_Location` be replaced with `CNS_region`?

jharenza commented:
@komalsrathi v2 is live

komalsrathi commented:
I have updated all modules with v2 except for `oncoplots` which I will do tomorrow after discussing on the call.

komalsrathi commented:
Hi! I think this PR is ready for review with the most up to date changes asked by @mkoptyra. Any chance of getting this reviewed and merged this week? If not then no worries. Thank you!

komalsrathi commented:
Any ETA on this PR? I would like to get this merged if possible - not sure what areas are lacking a review but please let me know. Thanks!

jharenza commented:
@komalsrathi go ahead and merge and we can more critically review the new issues as they are minor changes.

komalsrathi commented:
Thanks so much!
